acceptance/tests/pluginsync/4420_pluginfacts_should_be_resolvable_on_agent.rb

```diff
@@ -0,0 +1,91 @@
+test_name "Pluginsync'ed external facts should be resolvable on the agent"
+
+#
+# This test is intended to ensure that external facts downloaded onto an agent via
+# pluginsync are resolvable. In Linux, the external fact should have the same
+# permissions as its source on the master.
+#
+
+step "Create a codedir with a manifest and test module with external fact"
+codedir = master.puppet('master')['codedir']
+
+site_manifest_content =<<EOM
+node default {
+  include mymodule
+  notify { "foo is ${foo}": }
+}
+EOM
+
+unix_fact =<<EOM
+#!/bin/sh
+echo "foo=bar"
+EOM
+
+win_fact =<<EOM
+@echo off
+echo foo=bar
+EOM
+
+if agent['platform'] =~ /windows/
+  fact_content = win_fact
+else
+  fact_content = unix_fact
+end
+
+apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
+File {
+  ensure => directory,
+  mode   => "0755",
+  owner  => #{master.puppet['user']},
+  group  => #{master.puppet['group']},
+}
+
+file {
+  '#{codedir}':;
+  '#{codedir}/environments':;
+  '#{codedir}/environments/production':;
+  '#{codedir}/environments/production/manifests':;
+  '#{codedir}/environments/production/modules':;
+  '#{codedir}/environments/production/modules/mymodule':;
+  '#{codedir}/environments/production/modules/mymodule/manifests':;
+  '#{codedir}/environments/production/modules/mymodule/facts.d':;
+}
+
+file { '#{codedir}/environments/production/manifests/site.pp':
+  ensure => file,
+  content => '#{site_manifest_content}',
+}
+
+file { '#{codedir}/environments/production/modules/mymodule/manifests/init.pp':
+  ensure => file,
+  content => 'class mymodule {}',
+}
+
+file { '#{codedir}/environments/production/modules/mymodule/facts.d/external_fact.sh':
+  ensure  => file,
+  content => '#{fact_content}',
+}
+MANIFEST
+
+master_opts = {
+  'main' => {
+    'environmentpath' => "#{codedir}/environments"
+  }
+}
+
+step "Pluginsync the external fact to the agent and ensure it resolves correctly"
+with_puppet_running_on master, master_opts, codedir do
+  agents.each do |agent|
+    on(agent, puppet('agent', "-t --server #{master}"), :acceptable_exit_codes => [2])
```

Comment:
Second note: the `notify` in the test manifest used here causes Puppet to exit with `2`, so I made that the acceptable exit code.